tt_eager/tt_numpy/functions.hpp

```diff
@@ -352,6 +352,128 @@ static Tensor index_all(
     return output;
 }
 
+template<typename T>
+static Tensor fill_first_val_into_tensor(const Tensor& input_tensor, DataType data_type,
+			  const Layout layout , Device * device = nullptr,
+			  const MemoryConfig& output_mem_config = MemoryConfig{.memory_layout=tt::tt_metal::TensorMemoryLayout::INTERLEAVED}) {
+    const Shape& s_a = input_tensor.get_legacy_shape();
+    auto owned_buffer = tt_metal::owned_buffer::create<T>(tt_metal::compute_volume(s_a)); //ouput
+    auto device_buffer = input_tensor.device_buffer();
+    uint32_t size_in_bytes = device_buffer->size();
+    vector<T> data_vec;
+    const char *TT_METAL_SLOW_DISPATCH_MODE = std::getenv("TT_METAL_SLOW_DISPATCH_MODE");
+    if (TT_METAL_SLOW_DISPATCH_MODE == nullptr) {
+        data_vec.resize(size_in_bytes / sizeof(T));
+        tt::tt_metal::tensor_impl::read_data_from_device_buffer<T>(input_tensor.device()->command_queue(), device_buffer, data_vec.data(), true);
+    } else {
+        tt::tt_metal::tensor_impl::read_data_from_device_buffer<T>(device_buffer, data_vec);
+    }
+    auto input_buffer = owned_buffer::create<T>(std::move(data_vec));
+    const Shape input_tensor_strides = input_tensor.strides();
+    auto val = input_buffer[0];
+    for(uint32_t i = 0; i < s_a[0]; i++) {
+        for(int32_t j = 0; j < s_a[1]; j++) {
+            for(int32_t k = 0; k < s_a[2]; k++) {
+                for(int32_t l = 0; l < s_a[3]; l++) {
+                    auto input_index = l + input_tensor_strides[2] * k + input_tensor_strides[1] * j + input_tensor_strides[0] * i;
+                    owned_buffer[input_index] = val;
+                }
+            }
+        }
+    }
+    auto output = Tensor(OwnedStorage{owned_buffer}, s_a, data_type, layout).to(layout);
+    if (device != nullptr) {
+        output = output.to(device, output_mem_config);
+    }
+    return output;
+}
+
+template<typename T>
+static Tensor prod_result_computation_GS(const Tensor& input_tensor, DataType data_type,
```

Comment:
Could you explain the reason/difference between GS and WH?

Reply:
As mentioned in the file, In GS, the result is stored in the last 2R*32C .In case of WH_B0, the same number of elements need to be accessed, but the placement of values in this architecture is different. I had to access the late 4 rows and alternating columns ( I have depicted this [here](https://drive.google.com/file/d/1MkWnU0723KueYHs4llIgoTAWmkObkRRc/view?usp=sharing) )